### PR TITLE
HPCC-27636 Prevent an unecessary/wasteful 2nd call to activity reset()

### PR DESCRIPTION
**Pull request by jakesmith**
also remove a redundant/unused executeChild method

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by ghalliday — APPROVED** [review]
I think it is probably correct, but hard to follow for several reasons:
- jira doesn't contain details of the problem
- I think CGraphBase is used for both graphs and subgraphs, but it isn't clear from the function names which they apply to (or both?)
- Some comments/tracing within functions suggest it is processing a graph when it is actually processing a subgraph (e.g. GraphPrintLog("Processing graph");)

**Comment by jakesmith** [comment]
> * jira doesn't contain details of the problem

Agreed, have updated the JIRA with a description.

> * I think CGraphBase is used for both graphs and subgraphs, but it isn't clear from the function names which they apply to (or both?)

all CGraphBase (and derivatives) are subgraphs or childgraphs (inc. loop graphs)

Thor sees a [outer workflow] 'graph' as a job CJobBase and derivatives, though agree there's some overlap confusion when Thor first sees the job it is referenced by wuid+'graph name' - but it equated to a CJob* in Thor.

> * Some comments/tracing within functions suggest it is processing a graph when it is actually processing a subgraph (e.g. GraphPrintLog("Processing graph");)

They're all subgraphs (or child graphs), and I'm sure it's not consistent, however, changing it to 'subgraph', when it's a child might be misleading and not sure it is worth making it conditional.
